### DISCUSSION TITLE
Elsa recommended hardware specs?

### DISCUSSION THREAD

**Discussion by rboylesDev** [elsa-workflows/elsa-core#5274] [Design & tech]
I was curious if Elsa had any idea of rough required hardware specs. We are looking to run it in docker with containers and trying to estimate the CPU and memory we will need for a single Elsa node.

**Comment by glime-ai** [comment]
Elsa Workflows does not provide specific hardware requirements as these can vary greatly depending on the complexity and volume of workflows you intend to run. However, since Elsa is a .NET application, you can start with the general recommendations for running .NET applications in Docker containers and adjust based on your performance testing and requirements.

As a starting point, you might consider allocating:

- **CPU**: At least 1 core for a small workload, but you should benchmark your specific workflows to determine CPU load and scale accordingly.
- **Memory**: At least 1 GB of RAM as a baseline, with more memory allocated based on the complexity of your workflows and the number of concurrent executions.

It's important to monitor the performance and resource usage in a test environment that simulates your expected production workload. This will give you a better idea of the resources you'll need to allocate for a single Elsa node. Additionally, consider the scalability features of Docker and your orchestrator (like Kubernetes or Docker Swarm) to handle varying loads.

For production environments, it's also recommended to follow best practices for running ASP.NET Core applications in Docker, which include setting resource limits and ensuring that the container's environment is configured correctly.

Remember to refer to the official Elsa documentation and Docker best practices for more detailed guidance and to stay updated with any recommendations specific to Elsa as the project evolves.

This comment was generated by [Glime](https://glimelab.ai/?utm_source=dRbvTb7IhCSYiI&utm_medium=github).
